supervisor/core.py

```diff
@@ -55,6 +56,13 @@ def state(self, new_state: CoreState) -> None:
             )
         finally:
             self._state = new_state
+            if self.sys_core.state not in CLOSING_STATES:
+                self.sys_loop.call_soon_threadsafe(
+                    self.sys_loop.create_task,
+                    self.sys_homeassistant.websocket.async_supervisor_update_event(
+                        "info", {"state": new_state}
+                    ),
+                )
```

Comment:
Let's create a small helper on `self.sys_homeassistant.WebSocket.fire_websocket` which also handles errors with nice logging. Also, this setter is called from the event loop